tests/jest/BookReader/BookReaderPublicFunctions.test.js

```diff
@@ -1,68 +1,78 @@
 import BookReader from '@/src/BookReader';
+import { sleep } from '@/src/plugins/tts/utils';
+import sinon from 'sinon';
 
 beforeAll(() => {
-  global.alert = jest.fn();
+  global.alert = sinon.fake();
 });
 afterEach(() => {
-  jest.restoreAllMocks();
+  sinon.restore();
 });
 
 describe('BookReader.prototype.toggleFullscreen', ()  => {
   test('uses `isFullscreen` to check fullscreen state', () => {
-    const isFSmock = jest.fn();
-    isFSmock.mockReturnValueOnce(false);
+    const isFSmock = sinon.fake();
+    // isFSmock.mockReturnValueOnce(false);
     const br = new BookReader();
     br.mode = br.constMode1up;
-    br.enterFullscreen = jest.fn();
+    br.enterFullscreen = sinon.fake();
```

Comment:
Comment: using the function form like this lets sinon.restore() work correctly, but it doesn't super matter here cause we're instantiating a new BR in each function (yes!!!). But just thought I'd mention :) I've generally only used `spy`/`stub` ; but not sure what the difference is between those and `fake`.

```js
    const enterFullscreenStub = sinon.stub(br, 'enterFullscreen');
    ...
    expect(enterFullscreenStub.callCount).toEqual(1);
```

Reply:
keeping unresolved bc really good tip